packages/auth/src/core/strategies/passkey.test.ts

```diff
@@ -0,0 +1,314 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect, use } from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import * as sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+
+import { mockEndpoint } from '../../../test/helpers/api/helper';
+import { testAuth, TestAuth, testUser } from '../../../test/helpers/mock_auth';
+import * as mockFetch from '../../../test/helpers/mock_fetch';
+import { Endpoint } from '../../api';
+import { UserInternal } from '../../model/user';
+import { signInWithPasskey, enrollPasskey } from './passkey';
+import { UserCredentialImpl } from '../user/user_credential_impl';
+import { APIUserInfo } from '../../api/account_management/account';
+
+use(chaiAsPromised);
+use(sinonChai);
+
+describe('passkey', async () => {
+  let auth: TestAuth;
+  let user: UserInternal;
+
+  beforeEach(async () => {
+    auth = await testAuth();
+    user = testUser(auth, 'mock-uid', '', true);
+    mockFetch.setUp();
+
+    const mockUserCredentialImpl = {
+      _fromIdTokenResponse: sinon.stub().returns({
+        user,
+        operationType: 'mock-operation-type'
+      })
+    };
+
+    sinon
+      .stub(UserCredentialImpl, '_fromIdTokenResponse')
+      .callsFake(mockUserCredentialImpl._fromIdTokenResponse);
+  });
+
+  afterEach(() => {
+    sinon.restore();
+    mockFetch.tearDown();
+  });
+
+  it('should sign in with passkey', async () => {
+    if (typeof navigator === 'undefined') {
+      return;
+    }
+    const mockCredential: PublicKeyCredential = {
+      id: 'vnl7UPdoJi-ETJi6_obUejdIuc1eIXfUWa1rs5tmacQ',
+      type: 'public-key',
+      rawId: new Uint8Array([1, 2, 3]),
+      response: {
+        clientDataJSON: new Uint8Array([4, 5, 6]),
+        attestationObject: new Uint8Array([7, 8, 9])
+      }
+    } as unknown as PublicKeyCredential;
```

Comment:
It's because the `mockCredential` is not a complete `PublicKeyCredential` as it's missing many fields and methods that are not nessecry for this test.